### PR TITLE
Modernized ESProducers in RecoPixelVertexing/PixelLowPtUtilities

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

- Added consumes calls to all ESProducers
- Moved ClusterShapeHitFilterESProducer to the plugins directory

#### PR validation:

The code compiles.
This was a technical change so should not affect any results.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
please test

File: RecoPixelVertexing/PixelLowPtUtilities/plugins/ClusterShapeHitFilterESProducer.cc [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+/*****************************************************************************/
+void ClusterShapeHitFilterESProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<std::string>("PixelShapeFile");
+  desc.add<std::string>("PixelShapeFileL1");
+  desc.add<std::string>("ComponentName");
+  desc.add<bool>("doPixelShapeCut", true);
+  desc.add<bool>("doStripShapeCut", true);
```
Comment (Dr15Jones) [inline RecoPixelVertexing/PixelLowPtUtilities/plugins/ClusterShapeHitFilterESProducer.cc]: @perrotta I believe this is also not like #27589. Here the only defaults set are what was previously being set as defaults in the C++ code for tracked parameters by playing the `exists` game:
https://github.com/cms-sw/cmssw/blob/master/RecoPixelVertexing/PixelLowPtUtilities/src/ClusterShapeHitFilterESProducer.cc#L21-L22

So again, using the `fillDescriptions` is the proper way we want people to inject defaults to tracked parameters from within C++.

File: RecoPixelVertexing/PixelLowPtUtilities/plugins/ClusterShapeHitFilterESProducer.cc [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+#include "RecoTracker/Record/interface/CkfComponentsRecord.h"
+#include "RecoLocalTracker/SiStripClusterizer/interface/ClusterChargeCut.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+class ClusterShapeHitFilterESProducer : public edm::ESProducer {
+public:
+  ClusterShapeHitFilterESProducer(const edm::ParameterSet&);
```
Comment (perrotta) [inline RecoPixelVertexing/PixelLowPtUtilities/plugins/ClusterShapeHitFilterESProducer.cc]: Shouldn't be added also the headers for
```C++
#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
#include "FWCore/ParameterSet/interface/ParameterSet.h"
```

Comment (Dr15Jones) [reply RecoPixelVertexing/PixelLowPtUtilities/plugins/ClusterShapeHitFilterESProducer.cc]: done

**Comment by Dr15Jones** [comment]
Please test

**Comment by perrotta** [comment]
+1
- Technical
- Jenkins tests pass

**Comment by fabiocos** [comment]
+1

**Comment by makortel** [comment]
part of #26748

**Pull request merged**